### PR TITLE
enable custom providers headers option

### PR DESCRIPTION
**Pull request by tabaktoni**
## Motivation and Resolution
Enable configuring custom fetch headers.

## Usage related changes
Example
```
    const provider = new Provider({
        sequencer: {
            baseUrl: '',
            feederGatewayUrl: '',
            gatewayUrl: '',
            headers: {
                'Authorization': 'Basic ...=='
            }
        },
    });
```

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [x] Documented the changes
- [x] Updated the docs (www)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

File: www/docs/API/provider.md [hunk 1]
```diff
@@ -259,10 +259,12 @@ The options for the provider depend on the network. The structure of the options
 - options.**baseUrl** - Base URL of the network
 - options.**feederGatewayUrl** - Feeder Gateway Endpoint of the network
 - options.**gatewayUrl** - Gateway Endpoint
+- options.**headers** - [Optional] custom fecth headers
```
Comment (3alpha) [inline www/docs/API/provider.md:262]: Spelling error "fecth"

File: www/docs/API/provider.md [hunk 2]
```diff
@@ -396,6 +398,8 @@ Gets the transaction trace from a tx hash.
 `new starknet.RpcProvider(options)`
 
 - options.**nodeUrl** - Starknet RPC node url
+- options.**headers** - [Optional] custom fecth headers
```
Comment (3alpha) [inline www/docs/API/provider.md:401]: Again here "fecth"

File: www/docs/API/provider.md (hunk 1 above)
Comment (ivpavici) [reply www/docs/API/provider.md:262]:
```suggestion
- options.**headers** - [Optional] custom fetch headers
```

File: www/docs/API/provider.md [hunk 3]
```diff
@@ -259,10 +259,12 @@ The options for the provider depend on the network. The structure of the options
 - options.**baseUrl** - Base URL of the network
 - options.**feederGatewayUrl** - Feeder Gateway Endpoint of the network
 - options.**gatewayUrl** - Gateway Endpoint
+- options.**headers** - [Optional] custom fecth headers
 
 or
 
 - options.**network** - Either 'mainnet-alpha' or 'goerli-alpha'
+- options.**headers** - [Optional] custom fecth headers
```
Comment (ivpavici) [inline www/docs/API/provider.md:267]:
```suggestion
- options.**headers** - [Optional] custom fetch headers
```

File: www/docs/API/provider.md (hunk 2 above)
Comment (ivpavici) [inline www/docs/API/provider.md:401]:
```suggestion
- options.**headers** - [Optional] custom fetch headers
```

**Review by ivpavici — APPROVED** [review]

**Pull request merged**